code_generator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -9,493 +9,333 @@
-                var outgoingRequest = (<#= commandDetails.CommandName #>.OutgoingRequest) requestContext.Request;
-                var response = new <#= commandDetails.CommandName #>.Response(op.EntityId.Id, op.Message, (CommandStatusCode) op.StatusCode, unityResponsePayload, outgoingRequest.RawRequest);
-
-                view.AddCommandResponse(entity, response, <#= commandDetails.CommandName #>ResponsePool);
+                CommandResponses[commandIndex](op);
             }
 
-            private void Send<#= commandDetails.CommandName #>Request(Connection connection, <#= commandDetails.CommandName #>.OutgoingRequest outgoingRequest)
+            private void ApplyAuthorityChange(Unity.Entities.Entity entity, Authority authority, long entityId)
             {
-                var requestPayload = <#= commandDetails.RequestType #>.ToSpatial(outgoingRequest.RawRequest);
-                var request = new <#= componentDetails.FullyQualifiedSpatialTypeName #>.Commands.<#= commandDetails.CommandName #>.Request(requestPayload);
-
-                var requestId = connection.SendCommandRequest(new global::Improbable.EntityId(outgoingRequest.TargetEntityId), request, null);
+                switch (authority)
+                {
+                    case Authority.Authoritative:
+                        if (!entityManager.HasComponent<NotAuthoritative<<#= componentDetails.TypeName #>>>(entity)) 
+                        {
+                            LogDispatcher.HandleLog(LogType.Error, new LogEvent(InvalidAuthorityChange)
+                                .WithField(LoggingUtils.LoggerName, LoggerName)
+                                .WithField(LoggingUtils.EntityId, entityId)
+                                .WithField("New State", Authority.Authoritative)
+                                .WithField("Expected Old State", Authority.NotAuthoritative)
+                            );
+                            return;
+                        }
 
-                RequestIdToRequestContext.Add(requestId.Id, new RequestContext(outgoingRequest.SenderEntityId, outgoingRequest));
-            }
+                        entityManager.RemoveComponent<NotAuthoritative<<#= componentDetails.TypeName #>>>(entity);
+                        entityManager.AddComponentData(entity, new Authoritative<<#= componentDetails.TypeName #>>());
+                        break;
+                    case Authority.AuthorityLossImminent:
+                        if (!entityManager.HasComponent<Authoritative<<#= componentDetails.TypeName #>>>(entity)) 
+                        {
+                            LogDispatcher.HandleLog(LogType.Error, new LogEvent(InvalidAuthorityChange)
+                                .WithField(LoggingUtils.LoggerName, LoggerName)
+                                .WithField(LoggingUtils.EntityId, entityId)
+                                .WithField("New State", Authority.AuthorityLossImminent)
```

Comment:
New state isn't a great key. Authority would be better. Especially as there is only one transition you can make. So both bits of information aren't needed.